src/components/side-panel/layer-panel/panel-view-list-toggle.js

```diff
@@ -0,0 +1,110 @@
+import React, {useMemo} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import styled from 'styled-components';
+import OrderByList from 'components/common/icons/order-by-list';
+import OrderByDataset from 'components/common/icons/order-by-dataset';
+import {Tooltip} from 'components/common/styled-components';
+import {FormattedMessage} from 'localization';
+import {toggleLayerPanelListView} from 'actions/ui-state-actions';
+
+const PanelViewListToggleContainer = styled.div.attrs({
+  className: 'panel-view-list-toggle'
+})``;
+
+const PanelViewListToggleWrapper = styled.div.attrs({
+  className: 'panel-view-list-toggle-inner'
+})`
+  display: flex;
+  justify-content: flex-end;
+  align-content: center;
+  gap: 10px;
+`;
+
+export const StyledToggleOption = styled.div.attrs({
+  className: 'layer-panel-toggle-option'
+})`
+  color: ${props =>
+    props.active
+      ? props.theme.layerPanelToggleOptionColorActive
+      : props.theme.layerPanelToggleOptionColor};
+  :hover {
+    cursor: pointer;
+    color: ${props => props.theme.layerPanelToggleOptionColorActive};
+  }
+`;
+
+function ToggleOptionFactory() {
+  const ToggleOption = ({isActive, onClick, option}) => (
+    <StyledToggleOption
+      data-tip
+      data-for={`${option.id}-toggle-option`}
+      active={isActive}
+      onClick={onClick}
+    >
+      <option.iconComponent height="20px" />
+      <Tooltip id={`${option.id}-toggle-option`} effect="solid" delayShow={500} place="bottom">
+        <span>
+          <FormattedMessage id={option.label} />
+        </span>
+      </Tooltip>
+    </StyledToggleOption>
+  );
+
+  return ToggleOption;
+}
+
+const TOGGLE_OPTIONS = [
+  {
+    id: 'list',
+    iconComponent: OrderByList,
+    value: 'list',
+    label: 'List'
+  },
+  {
+    id: 'sort-by-dataset',
+    iconComponent: OrderByDataset,
+    value: 'sortByDataset',
+    label: 'Sort by dataset'
+  }
+];
+
+PanelViewListToggleFactory.deps = [ToggleOptionFactory];
+
+function PanelViewListToggleFactory(ToggleOption) {
+  const PanelViewListToggle = () => {
+    const layerPanelListViewMode = useSelector(
+      state => state.demo.keplerGl.map.uiState.layerPanelListView
```

Comment:
don't use `useSelector` inside kepler.gl, kepler reducer can be mounted at any app state path. the `state.demo.keplerGl` is the demo app path. and for other application kepler can be mounted at other state path. Here it's better to just pass `uiState.layerPanelListView` through KeplerGl -> SidePanel